instrumentation/akka-http-10.0/src/main/java/io/opentelemetry/auto/instrumentation/akkahttp/AkkaHttpClientInstrumentation.java

```diff
@@ -83,57 +81,49 @@ public AkkaHttpClientInstrumentation() {
 
   public static class SingleRequestAdvice {
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static SpanWithScope methodEnter(
-        @Advice.Argument(value = 0, readOnly = false) HttpRequest request) {
+    public static void methodEnter(
+        @Advice.Argument(value = 0, readOnly = false) HttpRequest request,
+        @Advice.Local("otelSpan") Span span,
+        @Advice.Local("otelScope") Scope scope,
+        @Advice.Local("otelCallDepth") Depth callDepth) {
       /*
       Versions 10.0 and 10.1 have slightly different structure that is hard to distinguish so here
       we cast 'wider net' and avoid instrumenting twice.
       In the future we may want to separate these, but since lots of code is reused we would need to come up
       with way of continuing to reusing it.
        */
-      int callDepth = CallDepthThreadLocalMap.incrementCallDepth(HttpExt.class);
-      if (callDepth > 0) {
-        return null;
+      callDepth = TRACER.getCallDepth();
```

Comment:
@trask Do we need this on client side even though we don't create duplicate client spans? Not something to fix in this PR though

Reply:
oh that's a really good point, and will reduce the (growing) size of our "instrumentation template"!

i'll open an issue for this